### PR TITLE
Auto confirm admin reactions

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Feature

## Description

Reactions by admins (roles "admin" and "super admin") will be automatically confirmed.

## Related Tickets & Documents

https://github.com/thepracticaldev/tech-private/issues/411

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

n/a

## Added tests?

- [X] yes

## Added to documentation?

- [X] no documentation needed

### CODE REVIEW

File: app/models/reaction.rb [hunk 1]
```diff
@@ -170,6 +170,8 @@ def permissions
   end
 
   def negative_reaction_from_untrusted_user?
+    return if user.any_admin?
```
Comment (citizen428) [inline app/models/reaction.rb]: I assume all admins are implicitly trusted, let me know if I should remove this guard.

**Comment by benhalpern** [comment]
This works for me. I do like the checks and balances that not doing this provided, but I still think this makes sense.

I think this puts more emphasis on closing the loop on mod action audit logs to pair with this to have better visibility into the goings on here.

File: app/models/reaction.rb (hunk 1 above)
Comment (jacobherrington) [inline app/models/reaction.rb]: `return if user&.any_admin?`

Will fix the failing spec.

**Comment by jacobherrington** [comment]
I'm interested in how this affects day-to-day quality of life for the people confirming these reports. I'm ignorant regarding who in prod is actually an admin, so I'm not sure what impact this will have on the reports.

@citizen428 wrote a spec for it, so we're clearly aware it is happening, but we might want to note explicitly that this works for vomits on users as well which is a much more destructive action than negative reactions and vomits on articles. I expect that is intended behavior, but in that specific use-case, it might be worth thinking about that idea of checks and balances that @benhalpern mentioned.

File: app/models/reaction.rb (hunk 1 above)
Comment (citizen428) [reply app/models/reaction.rb]: Thanks @jacobherrington. I had made the change locally (all specs run fine for me) but for some reason did not push it.

**Comment by citizen428** [comment]
> I do like the checks and balances that not doing this provided, but I still think this makes sense.

@benhalpern I specifically restricted this to admin and super-admins. Tag mods and trusted users still go through the approval process. And while things are auto-approved, they still create DB records, so we can still check this via Blazer or similar. 

> I'm interested in how this affects day-to-day quality of life for the people confirming these reports. 

@jacobherrington To quote a part of the original issue:

> Earlier today, I went through all of a user's comments (https://dev.to/<username>) and vomited on each one. Then, I needed to go to https://dev.to/internal/reports and confirm each one of the vomits manually.

This does not sound like a great experience for admins, especially if they try not to use the nuclear option of vomiting on the user.

**Pull request merged**